### PR TITLE
Replace rake-compiler-dev-box in favour of rake-compiler-dock

### PR DESCRIPTION
**Pull request by larskanis**
[Rake-compiler-dev-box](https://github.com/tjschuck/rake-compiler-dev-box) is no longer maintained by it's author. Although there are several forks, each adding changes for a given target project, it is generally more extensive to set up and to integrate. Vagrant images are built on the users computer and are not versioned. Moreover the vagrant images are mutable, so that every build can make persistent changes to the box. In the end this makes each and every rake-compiler-dev-box rather unique.

[Rake-compiler-dock](https://github.com/larskanis/rake-compiler-dock) comes as a gem and solves these issues by using Docker instead of Vagrant. That makes rake-compiler-dock faster and more reliable. It also provides a simple API that let's the user integrate the cross build into the Rakefile of the given project. In addition, it is also usable on Windows.

Although JRuby was included in the first step, it didn't make much sense. JRuby can easily be installed native on each target platform. So now, rake-compiler-dock focuses on Windows binary gems only. This fits to the given README section, too.

To sum up, I wouldn't recommend rake-compiler-dev-box to users of rake-compiler, any longer.

### CODE REVIEW

**Comment by kou** [comment]
Thanks for your suggestion. I've merged this pull request.

If you'll keep maintaining rake-compiler-dock, how about moving larskanis/rake-compiler-dock to rake-compiler/rake-compiler-dock? If we recommend rake-compiler-dock, it may be better. Users may feel relieved if rake-compiler-dock exists in the "official" organization.

**Comment by larskanis** [comment]
@kou Good point - I not even thought about that! But as a companion project to rake-compiler, it makes sense to place it there. I'll keep maintaining rake-compiler-dock in the future, because I use it for several [open source  projects](https://github.com/larskanis/rake-compiler-dock/wiki/Projects-using-rake-compiler-dock) and for some closed ones. What do I have to do for the migration?

**Comment by kou** [comment]
Thanks for accepting my suggestion.
I've invited you to "Owner" team of rake-compiler organization. The team has "admin" right.

Now, you can move larskanis/rake-compiler-dock to rake-compiler/rake-compiler-dock at https://github.com/larskanis/rake-compiler-dock/settings . You can find "Transfer ownership" item in the page.

**Comment by larskanis** [comment]
Seems that everything worked. Thank you! Old links are redirected, now, but I'll update all places I know about to the new location. And thank you for taking over the maintenance of rake-compiler!

**Comment by kou** [comment]
I also confirmed! :-)
